src/ui/contexts/TransactionsContext.tsx

```diff
@@ -31,58 +32,68 @@ export function TransactionsContextProvider({
     setTxs({ ...txs, [id]: { ...tx, status: TxStatusMap.Processing } });
-    const injector = systemChainType.isDevelopment ? undefined : await web3FromAddress(accountId);
-    const account = systemChainType.isDevelopment ? keyring.getPair(accountId) : accountId;
+
+    const keyPair = keyring.getPair(accountId);
+    let addressOrPair: AddressOrPair = keyPair;
+    let signer = undefined;
+
+    // If the account is not a testing account (//Alice etc.),
+    // we need the `signer` from the extension
+    if (!keyPair.meta.isTesting) {
+      signer = (await web3FromAddress(accountId)).signer;
+      // Only use plain address, otherwise pjs api want to sign with given KeyPair
+      // instead of `signer`
+      addressOrPair = keyPair.address;
+    }
 
     try {
-      const unsub = await extrinsic.signAndSend(
-        account,
-        { signer: injector?.signer || undefined },
-        async result => {
-          if (result.isInBlock || result.isFinalized) {
-            const events: Record<string, number> = {};
-
-            result.events.forEach(record => {
-              const { event } = record;
-              const key = `${event.section}:${event.method}`;
-              if (!events[key]) {
-                events[key] = 1;
-              } else {
-                events[key]++;
-              }
-            });
-
-            if (!isValid(result)) {
-              setTxs({ ...txs, [id]: { ...tx, status: TxStatusMap.Error, events } });
-
-              let message = 'Transaction failed';
-
-              if (result.dispatchError?.isModule) {
-                const decoded = api?.registry.findMetaError(result.dispatchError.asModule);
-                message = `${decoded?.section.toUpperCase()}.${decoded?.method}: ${decoded?.docs}`;
-              }
-
-              onError && onError(result);
-
-              throw new Error(message);
+      const unsubscribe = await extrinsic.signAndSend(addressOrPair, { signer }, async result => {
+        if (result.isInBlock || result.isFinalized) {
+          const events: Record<string, number> = {};
+
+          // Count events
+          result.events.forEach(record => {
+            const { event } = record;
+            const key = `${event.section}:${event.method}`;
+            if (!events[key]) {
+              events[key] = 1;
+            } else {
+              events[key]++;
             }
+          });
+
+          if (!isValid(result)) {
+            setTxs({ ...txs, [id]: { ...tx, status: TxStatusMap.Error, events } });
 
+            let message = 'Transaction failed';
+
+            if (result.dispatchError?.isModule) {
+              const decoded = api?.registry.findMetaError(result.dispatchError.asModule);
+              message = `${decoded?.section.toUpperCase()}.${decoded?.method}: ${decoded?.docs}`;
+            }
+
+            onError && onError(result);
+
+            unsubscribe();
+            throw new Error(message);
```

Comment:
Maybe we can provide a brief reason behind transaction failure?

Reply:
I lines 77-80 we try to decode the given error with the metadata we have. The raw `result` gets passed along to a potential error handler, `onError` and last but not least we throw a proper Error with a generic or decoded error message. Depending how the rest is implemented the error is caught or an `onError` provided, which then presents the error to the user somehow, either a toast popping up or a failure message. 

I think that's all we could do here. 🤔

Reply:
Ah, my bad, did not notice the error decoding